src/AstTransforms.cpp

```diff
@@ -221,6 +222,117 @@ bool UniqueAggregationVariablesTransformer::transform(AstTranslationUnit& transl
     return changed;
 }
 
+std::string MaterializeSingletonAggregationTransformer::findUniqueVariableName(const AstClause& clause) {
```

Comment:
The private static helpers could be limited to this cpp file. Just wrap the methods in
```
namespace {
local function
}
```
so that they can only be seen in this translation unit

Reply:
oh ok. so i am wrapping it in an anonymous namespace?

Reply:
I changed it to what I think you mean, is it indented right though?

Reply:
Wrapping it in an anonymous namespace, yes, but that only works if it's a local function - not if it's a class method. Either approach is arguable, especially since the 'local' version is shared with the other transformers in AstTransforms.cpp.

Sooo, stick with your original class version since you prefer that?

Reply:
Okay that makes sense. Honestly I thought it was a bit inappropriate to put it as a class method cuz its definitely not specific functionality for the class so maybe now i will change it to local so i feel like i have contributed to this process